tests/python_package_test/test_basic.py

```diff
@@ -332,6 +332,133 @@ def check_asserts(data):
+        np.testing.assert_equal(len(train_data_csr.get_feature_name()), expected_num_features)
+        np.testing.assert_equal(booster_csr.num_feature(), expected_num_features)
+        pred_csr = booster_csr.predict(sparse.csr_matrix(X_test))
+        pred_contrib_csr = booster_csr.predict(sparse.csr_matrix(X_test), pred_contrib=True)
+        if model_prefix == "multiclass":
+            pred_contrib_csr = np.hstack([csr.toarray() for csr in pred_contrib_csr])
+        else:
+            pred_contrib_csr = pred_contrib_csr.toarray()
+        np.testing.assert_allclose(pred_csr, pred_1)
+        np.testing.assert_allclose(pred_contrib_1, pred_contrib_csr)
+
+        # test that target encoding with csc format works
+        train_data_csc = lgb.Dataset(sparse.csc_matrix(X_train), label=y_train,
+                                     category_encoders=category_encoders_str)
+        valid_data_csc = lgb.Dataset(sparse.csc_matrix(X_test), label=y_test,
+                                     category_encoders=category_encoders_str, reference=train_data_csc)
+        booster_csc = lgb.train(params, train_data_csc, valid_sets=[valid_data_csc], valid_names=["valid_data"])
+        np.testing.assert_equal(train_data_csc.num_feature(), expected_num_features)
+        np.testing.assert_equal(valid_data_csc.num_feature(), expected_num_features)
+        np.testing.assert_equal(len(train_data_csc.get_feature_name()), expected_num_features)
+        np.testing.assert_equal(booster_csc.num_feature(), expected_num_features)
+        pred_csc = booster_csc.predict(sparse.csc_matrix(X_test))
+        pred_contrib_csc = booster_csc.predict(sparse.csc_matrix(X_test), pred_contrib=True)
+        if model_prefix == "multiclass":
+            pred_contrib_csc = np.hstack([csc.toarray() for csc in pred_contrib_csc])
+        else:
+            pred_contrib_csc = pred_contrib_csc.toarray()
+        np.testing.assert_allclose(pred_csc, pred_1)
+        np.testing.assert_allclose(pred_contrib_1, pred_contrib_csc)
+
+    X_train, X_test, y_train, y_test = train_test_split(*load_breast_cancer(return_X_y=True),
+                                                        test_size=0.1, random_state=2)
+
+    params = {
+        "objective": "binary",
+        "metric": "auc",
+        "min_data": 10,
+        "num_leaves": 15,
+        "verbose": 1,
+        "max_bin": 255,
+        "max_cat_to_onehot": 1
+    }
+
+    test_category_encoding_inner(tmp_path, X_train, X_test, y_train, y_test, params, "binary")
+
+    # test target encoding under multi-class case
+    X, y = load_iris(return_X_y=True)
+    # convert float to int, so that we can treat them as categorical features
+    X = np.array(np.array(X, dtype=np.int), dtype=np.float)
+    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
+    params = {
+        'objective': 'multiclass',
+        'metric': 'multi_logloss',
```

Comment:
Assert for this metric.

Reply:
Resolved via a25f323 by comparing the evaluation results.